src/app/forms/validators/custom-async-validators.ts

```diff
@@ -114,7 +117,62 @@ export class CustomAsyncValidators {
 
           return null;
         })
+   
+        );
+  }
+
+  static hideIfEqualsWithCondition(store: Store<State>, sibling: string, value: string, conditions: Condition | Condition[]): AsyncValidatorFn {
+    return (control: AbstractControl): Observable<ValidationErrors | null> =>
+      store.pipe(
+        take(1),
+        select(testResultInEdit),
+        map(testResult => {
+
+          if(testResult){
+            const conditionsPassed = CustomAsyncValidators.checkConditions(testResult, conditions);
+
+            if (control?.parent) {
+              const siblingControl = control.parent.get(sibling) as CustomFormControl;
+              siblingControl.meta.hide = Array.isArray(value) ? (conditionsPassed && value.includes(control.value)) : (conditionsPassed && control.value === value);
+            }
+          }
+
+          return null;
+        })
       );
   }
- 
+
+  static checkConditions(testResult: TestResultModel, conditions: Condition | Condition[] ){
+    if(Array.isArray(conditions)){
+      let failed: boolean = false;
+
+      conditions.forEach((condition) => {
+        if(!CustomAsyncValidators.checkCondition(testResult, condition)) failed = true;
+      });
+
+      return !failed;
+    }else{
+      return CustomAsyncValidators.checkCondition(testResult, conditions);
+    }
```

Comment:
This can be simplified.
```suggestion
    if (!Array.isArray(conditions)) {
      return CustomAsyncValidators.checkCondition(testResult, conditions);
    }

    for (const condition of conditions) {
      if(!CustomAsyncValidators.checkCondition(testResult, condition)) {
        return false;
      }
    }
    
    return true;
```

Reply:
Thanks Taha that looks simpler I will commit that.